mro/table.py

```diff
@@ -1,21 +1,34 @@
-﻿
+﻿from contextlib import contextmanager
+import logging
+import threading
+import time
+
+import psycopg2
+
 import mro.connection as con
 import mro.data_types
 import mro.foreign_keys
-import time
-import psycopg2
-import logging
-import threading
+
 
 logger = logging.getLogger(__name__)
 
 # TODO replace with green thread friendly, thread local storage
 psycopg2_lock = threading.Lock()
 
 
-class table(object):
+@contextmanager
+def disable_insert():
+    table._insert.disabled = True
```

Comment:
I think I'd have put it inside the try block but don't think it makes much difference.